src/main/java/com/mraof/minestuck/block/SpikeBlock.java

```diff
@@ -0,0 +1,80 @@
+package com.mraof.minestuck.block;
+
+import com.mraof.minestuck.entity.underling.UnderlingEntity;
+import com.mraof.minestuck.util.CustomVoxelShape;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.MobEntity;
+import net.minecraft.pathfinding.PathNodeType;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+
+public class SpikeBlock extends DecorBlock
+{
+	public SpikeBlock(Properties properties, CustomVoxelShape shape)
+	{
+		super(properties, shape);
+	}
+	
+	@Override
+	public void fallOn(World worldIn, BlockPos pos, Entity entityIn, float fallDistance)
+	{
+		if(entityIn instanceof UnderlingEntity)
+		{
+			entityIn.causeFallDamage(fallDistance, 1.5F); //damage reduced for underlings
```

Comment:
My feeling was to make spikes less effective as a instantaneous grist farming device and so that if it is employed in a dungeon that the spikes are less likely to kill underlings that accidentally fall in to it. Although this reasoning I made is debatable and I am more than willing to reverse this decision

Reply:
So while you did update the comment, it doesn't quite reflect what you wrote there. Anyways here are various thoughts on what could be changed:
* Underlings tend to have a lot more hp than regular mobs. Players also get an additional damage bonus against underlings from rungs. With those combined, there is already "some" measure against non-player damage. With that background, you could remove this special case. (In addition, it could also make sense to leave it out, but keep this in mind during playtesting to identify if this is actually meaningful to have)
If that doesn't sound good and you still want to keep it:
* If the goal is "to limit their death by factors other than players", it makes more sense to simply add resistances to damages by reducing damage based on the damage source in the UnderlingEntity class.
* You could update the comment to tell that this is meant as an anti-grist-farm measure, and try to better tell what the end goal is.
* If you're only changing the damage multiplier based on the entity type, I'd suggest separating it a bit more from the function that deals the damage. The reason being to only define what type of damage is used (the damage source) in one place, instead of two.
* You could use the underlings entity tag instead of a call to instanceof. Would make it less hardcoded to a specific java class, and tags seems ideal for this purpose.